src/awkward/_connect/pyarrow.py

```diff
@@ -903,6 +925,9 @@ def is_revertable(akarray):
 
 
 def remove_optiontype(akarray):
+    if not is_revertable(akarray):
+        # In the case of the Unknown type, we get an EmptyArray at this point.
+        return akarray
```

Comment:
I did ... something here although it feels a little scary. Curious what you think!

While I did not really like my `if not is_revertable` line, in a very limited way it seems reasonable: the very next line assumes akarray has a __pyarrow_original property even though that's not a property of Content. Things that prevent crashes seem wise, *if* there is a clearly correct way to handle that case. It felt like maybe I was doing just that.

Still, with my addition of `revertable(out, out)` (again, cringe?) this check seems okay to remove.

Reply:
`revertable(out, out)` is fine: it's revertable to itself. When/if the reverting process comes along, it will get replaced by itself (i.e. nothing happens).

The value here is that now we have trees in which either every node has a `__pyarrow_original` or every nod does not have a `__pyarrow_original`, and so the code that removes these things doesn't have to be special-cased.

This is just a matter of keeping like with like.